src/webots/nodes/WbAbstractAppearance.cpp

```diff
@@ -37,6 +40,23 @@ WbAbstractAppearance::WbAbstractAppearance(const WbNode &other) : WbBaseNode(oth
   init();
 }
 
+WbAbstractAppearance::WbAbstractAppearance(const QString &modelName, const aiMaterial *material) : WbBaseNode(modelName, NULL) {
+  mInitializedFromAssimpMaterial = true;
+
+  aiString name("PBRAppearance");
+  material->Get(AI_MATKEY_NAME, name);
+  mName = new WbSFString(QString(name.C_Str()));
+  mNameValue = mName->value();
+
+  mTextureTransform = new WbSFNode(NULL);
+}
+
+WbAbstractAppearance::~WbAbstractAppearance() {
+  delete mName;
+  if (mTextureTransform)
+    delete mTextureTransform;
```

Comment:
It is OK to call delete with a NULL argument (it will simply do nothing):
```suggestion
  delete mTextureTransform;
```